src/Core/Framework/Context.php

```diff
@@ -41,20 +36,17 @@ class Context extends Struct
     protected $extensions = [];
 
     /**
-     * @param array<string> $languageIdChain
-     * @param array<string> $ruleIds
+     * @param string[] $ruleIds
```

Comment:
please use the "generic" notation for array types as it is more precise. `string[]` can be `array<string, string>`, `array<int, string>` or `list<string>`